### PR TITLE
move complexity exemplars to configstore with backfill migration

### PR DESCRIPTION
**Pull request by Madhuvod**
## Summary

Moves the curated semantic-routing exemplars out of the `complexity` plugin and into `configstore` so that both the governance runtime and the database migration layer share a single authoritative source. A new migration backfills those exemplars into persisted complexity configurations that were created before the defaults existed, without overwriting any administrator-owned phrases or tier assignments.

## Changes

- Relocated `exemplars.go` from `plugins/governance/complexity` to `framework/configstore`, changing its package and replacing the private `sharedTierDefaults` helper with a public `DefaultComplexityExemplars()` function that returns independent copies of all three tier slices via `cloneComplexityExemplars`.
- `sharedTierDefaults` moved back into `plugins/governance/complexity/config.go` where it is still needed to combine static keywords with exemplars.
- `DefaultEditableKeywordConfig` and `defaultFullKeywordConfig` in the complexity plugin now call `configstore.DefaultComplexityExemplars()` instead of referencing the package-local exemplar slices directly.
- Added `migrationBackfillDefaultComplexityExemplars`, which reads the persisted `ComplexityAnalyzerConfig`, appends any missing default exemplars per tier (existing phrases always win, cross-tier duplicates are not re-added), clears `EmbeddingFingerprint` to trigger re-embedding, validates, and writes back. The rollback is intentionally a no-op to avoid removing administrator data.
- `appendMissingDefaultComplexityExemplars` deduplicates by normalized key (lowercased, collapsed whitespace) across all tiers before appending, so a phrase already present in any tier is never duplicated.
- Added `framework/configstore/exemplars_test.go` covering: exact count per tier (50 each), global uniqueness, deep-copy isolation, word-length overlap between adjacent tiers, and surface-feature balance (question form, all-lowercase, embedded code) to prevent the classifier from learning tier identity from formatting alone.
- Added `TestMigrationBackfillDefaultComplexityExemplars` covering: backfill with preserved administrator state, legacy keyword field canonicalization, absent config, empty config, malformed config, idempotency on a complete config, and configs exceeding 500 phrases.

## Type of change

- [ ] Bug fix
- [ ] Feature
- [x] Refactor
- [ ] Documentation
- [ ] Chore/CI

## Affected areas

- [x] Core (Go)
- [ ] Transports (HTTP)
- [ ] Providers/Integrations
- [x] Plugins
- [ ] UI (React)
- [ ] Docs

## How to test

```sh
go test ./framework/configstore/... ./plugins/governance/complexity/...
```

The exemplar tests assert count, uniqueness, deep-copy safety, length-overlap between tiers, and surface-feature distribution. The migration tests run against an in-memory SQLite database and cover all edge cases including idempotency and legacy field canonicalization.

## Screenshots/Recordings

N/A

## Breaking changes

- [ ] Yes
- [x] No

## Related issues

N/A

## Security considerations

No auth, secrets, or PII involved. The migration only appends to existing rows and never creates new configuration entries.

## Checklist

- [ ] I read `docs/contributing/README.md` and followed the guidelines
- [x] I added/updated tests where appropriate
- [ ] I updated documentation where needed
- [x] I verified builds succeed (Go and UI)
- [ ] I verified the CI pipeline passes locally if applicable

### CODE REVIEW

**Comment by Madhuvod** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.com/github/pr/maximhq/bifrost/5842?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#5896** <a href="https://app.graphite.com/github/pr/maximhq/bifrost/5896?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 2  dependent PRs ([#5901](https://github.com/maximhq/bifrost/pull/5901) <a href="https://app.graphite.com/github/pr/maximhq/bifrost/5901?utm_source=stack-comment-icon" target="_blank">Graphite</a>, [#5924](https://github.com/maximhq/bifrost/pull/5924) <a href="https://app.graphite.com/github/pr/maximhq/bifrost/5924?utm_source=stack-comment-icon" target="_blank">Graphite</a>)
* **#5865** <a href="https://app.graphite.com/github/pr/maximhq/bifrost/5865?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#5842** <a href="https://app.graphite.com/github/pr/maximhq/bifrost/5842?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.com/github/pr/maximhq/bifrost/5842?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#5810** <a href="https://app.graphite.com/github/pr/maximhq/bifrost/5810?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#5791** <a href="https://app.graphite.com/github/pr/maximhq/bifrost/5791?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#5790** <a href="https://app.graphite.com/github/pr/maximhq/bifrost/5790?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#5743** <a href="https://app.graphite.com/github/pr/maximhq/bifrost/5743?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#5748** <a href="https://app.graphite.com/github/pr/maximhq/bifrost/5748?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#5742** <a href="https://app.graphite.com/github/pr/maximhq/bifrost/5742?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#5686** <a href="https://app.graphite.com/github/pr/maximhq/bifrost/5686?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#5680** <a href="https://app.graphite.com/github/pr/maximhq/bifrost/5680?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#5656** <a href="https://app.graphite.com/github/pr/maximhq/bifrost/5656?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#5655** <a href="https://app.graphite.com/github/pr/maximhq/bifrost/5655?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#5628** <a href="https://app.graphite.com/github/pr/maximhq/bifrost/5628?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#5602** <a href="https://app.graphite.com/github/pr/maximhq/bifrost/5602?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#5530** <a href="https://app.graphite.com/github/pr/maximhq/bifrost/5530?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#5529** <a href="https://app.graphite.com/github/pr/maximhq/bifrost/5529?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#5499** <a href="https://app.graphite.com/github/pr/maximhq/bifrost/5499?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#5497** <a href="https://app.graphite.com/github/pr/maximhq/bifrost/5497?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#5494** <a href="https://app.graphite.com/github/pr/maximhq/bifrost/5494?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `dev`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: framework/configstore/migrations.go [hunk 1]
```diff
@@ -11107,3 +11108,113 @@ func migrationAddPricingOverrideUserIDColumn(ctx context.Context, db *gorm.DB, l
...
+				return fmt.Errorf("persist complexity analyzer exemplar backfill: updated %d rows, want 1", result.RowsAffected)
+			}
+
+			logger.Info("[configstore] %s: added %d default complexity exemplars", migrationName, added)
+			return nil
+		},
+		Rollback: func(*gorm.DB) error {
+			// Removing phrases later would also remove administrator-owned data
+			// that happens to match a default exemplar.
+			return nil
```
[the remark this answers is not in the document]
Comment (Madhuvod) [reply framework/configstore/migrations.go]: this is fixed i think check
